lib/meta-feed-helpers.js

```diff
@@ -0,0 +1,242 @@
+const { SecretKey } = require('ssb-private-group-keys')
+const {
+  where,
+  and,
+  count,
+  isDecrypted,
+  type,
+  author,
+  toCallback,
+  toPullStream,
+} = require('ssb-db2/operators')
+const { fromMessageSigil } = require('ssb-uri2')
+const {
+  keySchemes,
+  validator: {
+    group: { init: initSpec },
+  },
+} = require('private-group-spec')
+const pull = require('pull-stream')
+const paraMap = require('pull-paramap')
+const clarify = require('clarify-error')
+
+function secretKeyFromPurpose(purpose) {
+  return new SecretKey(Buffer.from(purpose, 'base64'))
+}
+
+module.exports = (ssb) => {
+  function findOrCreateAdditionsFeed(cb) {
+    const details = {
+      purpose: 'group/additions',
+      feedFormat: 'classic',
+    }
+
+    ssb.metafeeds.findOrCreate(details, cb)
+  }
+
+  function findEmptyGroupFeed(rootId, cb) {
+    let found = false
+    pull(
+      ssb.metafeeds.branchStream({ root: rootId, old: true, live: false }),
+      pull.filter((branch) => branch.length === 4),
+      pull.map((branch) => branch[3]),
+      // only grab feeds that look like group feeds
+      pull.filter((feed) => feed.recps && feed.purpose.length === 44),
+      paraMap((feed, cb) => {
+        ssb.db.query(
+          where(author(feed.id)),
+          count(),
+          toCallback((err, count) => {
+            // prettier-ignore
+            if (err) return cb(clarify(err, 'Failed to count messages in group feed'))
+
+            if (count === 0) {
+              // we're only interested in empty feeds
+              return cb(null, feed)
+            } else {
+              return cb()
+            }
+          })
+        )
+      }, 4),
```

Comment:
I also added a `width` to this paramap (without width, paraMap behaves like `pull.async` , i.e. no parallel boost)